### ISSUE TITLE
[Tracker] Workaround MSBuild bugs

### ISSUE THREAD

**Issue by natemcmaster** [aspnet/KoreBuild#150]
Similar to this issue: https://github.com/aspnet/Mvc/issues/5482, except that here I'm tracking the list of bugs we have worked around to implement VS 2017 support.

Check off when the workaround has been removed.

**Resolution unknown**

- [ ] https://github.com/NuGet/Home/issues/3953 - dotnet pack - ProjectReference loses version information
- [x] ~~https://github.com/dotnet/sdk/issues/490 - Generate AssemblyMetadataAttribute when serviceable is true~~ won't fix. We resolved in our targets.
- [ ] https://github.com/NuGet/Home/issues/4254 -Support xplat command line for packing nuspec

**Should be resolved when upgrading to rc3**

- [x] https://github.com/NuGet/Home/issues/4063 - dotnet pack should pack tools with the versions that the package was built against. - Workaround removed in https://github.com/aspnet/BuildTools/pull/149
- [x]  https://github.com/Microsoft/vstest/issues/287 - OutputType=Exe required for .NET Core, but OutputType=Library required for .NET Framework test apps. Requires upgrade to Microsoft.NET.Test.Sdk 15.0.0-preview-20170106-0. orkaround removed in https://github.com/aspnet/BuildTools/pull/149

**Should be resolved when upgrading to RTM**

- [x]  https://github.com/NuGet/Home/issues/4321 Cannot pack .NET Core CLI tool
- [x] https://github.com/dotnet/sdk/issues/475 - build in parallel breaks badly (our workaround is to build one csproj at a time)
- [x] Microsoft/msbuild#1587 - Solution targets overriden by metaproj. (Workaround added in https://github.com/aspnet/Universe/pull/472)
- [x] dotnet/sdk#686 - runtimeconfig.json has incorrect version
- [x] https://github.com/Microsoft/vstest/issues/311 Tests run from wrong folder. Workaround added to Hosting, Razor, and others.
- [x] https://github.com/Microsoft/vstest/issues/283 - dotnet-test returns 0 on multi-tfm test projects

**Relevant PRs to track:**

- [x] https://github.com/dotnet/cli/pull/5214 - update CLI to NuGet RC3
- [x] https://github.com/dotnet/cli/pull/5234 - update CLI to latest VSTest
- [x] https://github.com/Microsoft/vstest/pull/310 - Fixes Microsoft/vstest#287
- [x] https://github.com/NuGet/NuGet.Client/pull/1135 - Fixes NuGet/Home#4321
- [x] https://github.com/Microsoft/msbuild/pull/1590- fixes Microsoft/msbuild#1587

**Comment by natemcmaster** [comment]
We've removed workarounds.
